provider_shopper/lib/screens/catalog.dart

```diff
@@ -0,0 +1,91 @@
+import 'package:flutter/material.dart';
+import 'package:provider/provider.dart';
+import 'package:provider_shopper/models/cart.dart';
+import 'package:provider_shopper/models/catalog.dart';
+
+class MyCatalog extends StatelessWidget {
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      body: CustomScrollView(
+        slivers: [
+          _MyAppBar(),
+          SliverToBoxAdapter(child: SizedBox(height: 12)),
+          SliverList(
+            delegate: SliverChildBuilderDelegate(
+                (context, index) => _MyListItem(index)),
+          ),
+        ],
+      ),
+    );
+  }
+}
+
+class _AddButton extends StatelessWidget {
+  final Item item;
+
+  const _AddButton({Key key, @required this.item}) : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    var cart = Provider.of<CartModel>(context);
+
+    return FlatButton(
+      onPressed: cart.items.contains(item) ? null : () => cart.add(item),
+      splashColor: Theme.of(context).primaryColor,
+      child: cart.items.contains(item)
+          ? Icon(Icons.check, semanticLabel: 'ADDED')
+          : Text('ADD'),
+    );
+  }
+}
+
+class _MyAppBar extends StatelessWidget {
+  @override
+  Widget build(BuildContext context) {
+    return SliverAppBar(
+      title: Text('Catalog', style: Theme.of(context).textTheme.display4),
+      floating: true,
+      actions: [
+        IconButton(
+          icon: Icon(Icons.shopping_cart),
+          onPressed: () => Navigator.pushNamed(context, '/cart'),
+        ),
+      ],
+    );
+  }
+}
+
+class _MyListItem extends StatelessWidget {
+  final int index;
+
+  _MyListItem(this.index) : super(key: ValueKey(index));
+
+  @override
+  Widget build(BuildContext context) {
+    var catalog = Provider.of<CatalogModel>(context);
+    var item = catalog.getByPosition(index);
+    var textTheme = Theme.of(context).textTheme.title;
+
+    return Padding(
+      padding: const EdgeInsets.symmetric(horizontal: 16, vertical: 8),
+      child: LimitedBox(
+        maxHeight: 48,
+        child: Row(
+          children: [
+            AspectRatio(
+              aspectRatio: 1,
+              child: Container(
+                color: item.color,
+              ),
+            ),
+            SizedBox(width: 24),
+            Text(item.name, style: textTheme),
```

Comment:
Unlikely to matter for your app, but I believe that unless the `Text` is wrapped in a `Flexible` (like `Expanded`), it will never break a line and could overflow to the right. Consider removing the `Spacer` and wrapping `Text` in an `Expanded` instead.

Reply:
Damn, you're right, of course. I was so enamored with Spacer (a new find for me) that I _removed_ Expanded and put a Spacer here, thinking how clever I was. 🤦‍♂